### PR TITLE
PLAT-2208 Abacship - fix random board test error

### PR DESCRIPTION
**Pull request by elizabethhealy**

### CODE REVIEW

File: examples/abacship-app/backend_server/app/game.py [hunk 1]
```diff
@@ -324,6 +324,8 @@ def genRandomBoard():
     board = [[OCEAN for _ in range(SIZE)] for _ in range(SIZE)]
     for ship in SHIPS:
         board = _placeRandom(board, ship)
+    while not validBoard(board):
```
Comment (dmihalcik-virtru) [inline examples/abacship-app/backend_server/app/game.py:327]:
```suggestion
    if not validBoard(board):
```

**Review by dmihalcik-virtru — CHANGES_REQUESTED** [review]
you only have to check each board once

File: examples/abacship-app/backend_server/app/game.py (hunk 1 above)
Comment (elizabethhealy) [reply examples/abacship-app/backend_server/app/game.py:327]: @dmihalcik-virtru wanted to prepare for the case where genBoard gives a wrong board twice in a row (very unlikely odds-wise but still possible)

Comment (dmihalcik-virtru) [reply examples/abacship-app/backend_server/app/game.py:327]: Since it is recursive, it will only ever return a valid board.

Comment (elizabethhealy) [reply examples/abacship-app/backend_server/app/game.py:327]: this solution is really just a work around, since genboard() is only used in the unit tests i didnt want to devote time to finding the edge case / off by one error

Comment (elizabethhealy) [reply examples/abacship-app/backend_server/app/game.py:327]: oops wait forgot to push the changes

**Review by dmihalcik-virtru — APPROVED** [review]
